colcon_core/shell/__init__.py

```diff
@@ -50,11 +50,23 @@ class ShellExtensionPoint:
     `sh` shell extension to setup environment variables and only contributes
     additional information like completion.
 
-    All "non-primiry" shell extensions must use a priority equal to or lower
+    All "non-primary" shell extensions must use a priority equal to or lower
     than the default.
     """
     PRIORITY = 100
 
+    def get_file_extensions(self):
+        """
+        Get the file extensions provided by this extension.
+
+        By default the extension name will be returned.
```

Comment:
By default the *shell name* will be returned ?

Reply:
The name of the extension will be returned - as specified for the entry point.

Reply:
yeah it just seem more prone to confusion to refer to the "file extension" or the "colcon extension" that's why I suggested a different wording